index.js

```diff
@@ -546,7 +546,14 @@ prototype.buildOperation = function(oldOperation, operationDefaults) {
   var tags = (operationDefaults.tags || []).concat(oldOperation.tags || []);
   tags = removeDuplicates(tags);
 
-  return extend({}, operationDefaults, {
+  var customProperties = {};
+  for (var key in oldOperation) {
+    if (/^x-/.test(key)) {
+      customProperties[key] = oldOperation[key];
+    }
+  }
+
+  return extend({}, operationDefaults, customProperties, {
```

Comment:
@bobvanderlinden Thanks for PR 👍 
Looks good as is.
But if you have the time it would be great to extract it into:
```js
  return extend({}, operationDefaults, getCustomProperties(oldOperation), {
  // ...
  });
}

function getCustomProperties(object) {
   var customProperties = {};
   // ...
}
```
So it would be easy to support custom properties in other places.

Reply:
Good suggestion and thanks for the quick review :+1: Added getCustomProperties.